src/ts/src/channel_log.ts

```diff
@@ -0,0 +1,85 @@
+/*------------------------------------------------------------------------------
+ * MIT License
+ *
+ * Copyright (c) 2021 IBM
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *----------------------------------------------------------------------------*/
+
+// Local
+import { AlogCoreSingleton } from './core';
+
+/**
+ * The ChannelLog object binds a channel name and wraps all of the core
+ * functions which are scoped to an individual channel.
+ */
+export class ChannelLog {
+
+  public channel: string;
+  private coreInstance: any;
```

Comment:
```suggestion
  private coreInstance: AlogCoreSingleton;
```

You shouldn't need `any` here or the `as any` below, should just be happy.

Reply:
Sadly, it's not. This is what I tried originally, but because of how [the functions are programmatically added to the singleton](https://github.com/IBM/alchemy-logging/blob/main/src/ts/src/core.ts#L119) Typescript is not happy with this.